### PR TITLE
update refund with unittest

### PR DESCRIPTION
**Pull request by ankitdas13**
for edit : https://github.com/razorpay/razorpay-node/blob/master/documents/refund.md#update-the-refund
Link : https://razorpay.com/docs/api/refunds/#update-refund

### CODE REVIEW

File: test/razorpay/test_refund.rb [hunk 1]
```diff
@@ -36,7 +36,21 @@ def test_create_refund
       stub_post(/refunds$/, 'fake_refund', "payment_id=#{@payment_id}")
       refund = Razorpay::Refund.create(payment_id: @payment_id)
       assert_instance_of Razorpay::Refund, refund
-      assert_equal refund.payment_id, @payment_id
+      #assert_equal refund.payment_id, @payment_id
```
Comment (arjunmenon) [inline test/razorpay/test_refund.rb:39]: Uncomment assertions

File: lib/razorpay/refund.rb [hunk 2]
```diff
@@ -19,5 +19,9 @@ def self.all(options = {})
     def self.fetch(id)
       request.fetch id
     end
+
+    def self.edit(id, options = {})
+      request.patch id, options
+    end
```
Comment (arjunmenon) [inline lib/razorpay/refund.rb:25]: Not an idiomatic way to implement this. Edit should work on an instance of refund obj. Pls see if we can rework on sdk arch when possible.

**Review by arjunmenon — COMMENTED** [review]
Pls add func docs as well

File: test/razorpay/test_refund.rb (hunk 1 above)
Comment (ankitdas13) [reply test/razorpay/test_refund.rb]: Done

File: lib/razorpay/refund.rb (hunk 2 above)
Comment (ankitdas13) [reply lib/razorpay/refund.rb]: yes this correct now ?
`para_attr = {
  "notes": {
    "notes_key_1": "Beam me up Scotty.",
    "notes_key_2": "Engage"
  }
}

Razorpay::Refund.fetch(refundId).edit(para_attr)`

File: documents/refund.md [hunk 3]
```diff
@@ -0,0 +1,307 @@
...
+
+### Create an instant refund
+
+```rb
+Razorpay::QrCode.create(paymentId,{
+  "amount": "100",
+  "speed": "optimum",
+  "receipt": "Receipt No. 31"
+})
+```
```
Comment (arjunmenon) [inline documents/refund.md:62]: Not part of `lib/razorpay/refund.rb`. And why `Razorpay::QrCode` to raise a instant refund request?

File: documents/refund.md [hunk 4]
```diff
@@ -0,0 +1,307 @@
...
+  "status": "processed",
+  "speed_requested": "optimum"
+}
+```
+-------------------------------------------------------------------------------------------------------
+
+### Fetch multiple refunds for a payment
+
+```js
+instance.payments.fetch_multiple_refund(paymentId,option)
```
Comment (arjunmenon) [inline documents/refund.md:100]: Pls update SDK docs correctly

File: documents/refund.md [hunk 5]
```diff
@@ -0,0 +1,307 @@
...
+      "speed_requested": "optimum"
+    }
+  ]
+}
+ ```
+-------------------------------------------------------------------------------------------------------
+
+### Fetch a specific refund for a payment
+```js
+instance.payments.fetch(paymentId,refundId)
```
[the remark this answers is not in the document]
Comment (ankitdas13) [reply documents/refund.md]: i have updated the doc 
doc reference : https://github.com/razorpay/razorpay-node/blob/master/documents/refund.md

File: documents/refund.md [hunk 6]
```diff
@@ -96,8 +103,12 @@ Razorpay::QrCode.create(paymentId,{
...
 ### Fetch multiple refunds for a payment
 
-```js
-instance.payments.fetch_multiple_refund(paymentId,option)
+```rb
+paymentId = "pay_FIKOnlyii5QGNx"
+
+option = {"count":1}
+
+Razorpay::Payment.fetch_multiple_refund(paymentId,option)
```
Comment (arjunmenon) [inline documents/refund.md:111]: Where is this declared?

File: documents/refund.md [hunk 7]
```diff
@@ -141,8 +152,12 @@ instance.payments.fetch_multiple_refund(paymentId,option)
...
 
 ### Fetch a specific refund for a payment
-```js
-instance.payments.fetch(paymentId,refundId)
+```rb
+paymentId = "pay_FIKOnlyii5QGNx"
+
+refundId = "rfnd_FP8DDKxqJif6ca"
+
+Razorpay::Payment.fetch(paymentId).fetch_refund(refundId)
```
Comment (arjunmenon) [inline documents/refund.md:160]: Where is this declared?

**Review by arjunmenon — COMMENTED** [review]
Why merge to [addon_update](https://github.com/razorpay/razorpay-ruby/tree/addon_update)?

File: documents/refund.md (hunk 6 above)
Comment (ankitdas13) [reply documents/refund.md:111]: its declared in payment module 

https://github.com/razorpay/razorpay-ruby/blob/payment_update/lib/razorpay/payment.rb#L42

File: documents/refund.md (hunk 7 above)
Comment (ankitdas13) [reply documents/refund.md:160]: its declared in payment module 
https://github.com/razorpay/razorpay-ruby/blob/payment_update/lib/razorpay/payment.rb#L38

**Comment by ankitdas13** [comment]
> Why merge to [addon_update](https://github.com/razorpay/razorpay-ruby/tree/addon_update)?

addon_update is sub-branch . while releasing we merge all pr with `addon_update` and test once again and then merge with master

File: documents/refund.md (hunk 6 above)
Comment (arjunmenon) [reply documents/refund.md:111]: Part of separate branch. Pls describe this function in the correct place.

File: documents/refund.md (hunk 7 above)
Comment (arjunmenon) [reply documents/refund.md:160]: Part of separate branch. Pls describe this function in the correct place.

File: documents/refund.md (hunk 6 above)
Comment (ankitdas13) [reply documents/refund.md:111]: according to [doc](https://razorpay.com/docs/api/refunds/#fetch-multiple-refunds-for-a-payment) this function is belongs to refund module that why i have added this function in refund.md

File: documents/refund.md (hunk 7 above)
Comment (ankitdas13) [reply documents/refund.md:160]: according to [doc](https://razorpay.com/docs/api/refunds/#fetch-a-specific-refund-for-a-payment) this function is belongs to refund module that why i have added this function in refund.md

**Comment by arjunmenon** [comment]
> addon_update is sub-branch . while releasing we merge all pr with `addon_update` and test once again and then merge with master

Do not agree with this approach. Changes should have gone in independently. `addon_update` was already reviewed once, if you are merging all other Prs there, would not be able to look into them again. Pls ensure stability of your changes. Approving for now.

**Review by arjunmenon — APPROVED** [review]
Approving for now.

**Pull request merged**